easypost/webhook.py

```diff
@@ -17,3 +27,30 @@ def update(self, **params) -> "Webhook":
         response, api_key = requestor.request(method=RequestMethod.PATCH, url=url, params=params)
         self.refresh_from(values=response, api_key=api_key)
         return self
+
+    @staticmethod
+    def validate_webhook(webhook_body: Dict[str, Any], headers: Dict[str, Any], webhook_secret: str) -> bool:
+        """Validate a webhook originated from EasyPost by comparing the HMAC header to a shared secret.
+        This function will return True if the webhook originated from EasyPost and False if it did not.
+        """
+        secret_validated = False
+        easypost_signature = headers.get("x-hmac-signature")
+
+        if easypost_signature:
+            normalized_secret = unicodedata.normalize("NFKD", webhook_secret)
+            encoded_secret = bytes(normalized_secret, "utf8")
+            hexified_secret = binascii.hexlify(encoded_secret)
+            webhook_bytes = json.dumps(webhook_body).encode()
+
+            expected_signature = hmac.new(
+                key=hexified_secret,
+                msg=webhook_bytes,
+                digestmod=hashlib.sha256,
+            )
+
+            digest = "hmac-sha256-hex=" + expected_signature.hexdigest()
+
+            if hmac.compare_digest(digest, easypost_signature):
+                secret_validated = True
```

Comment:
generally my preference for an API design is to raise an exception on mismatch, instead of returning a boolean. That makes it harder to do the insecure thing.

Reply:
Doh, that would be the natural thing wouldn't it? Thanks for the callout. It now raises an exception instead.